test/builtin_dataset_mocks.py

```diff
@@ -30,13 +29,11 @@
 
 
 class DatasetMock:
-    def __init__(self, name, mock_data_fn):
-        self.dataset = find(name)
-        self.info = self.dataset.info
-        self.name = self.info.name
-
+    def __init__(self, name, *, mock_data_fn, configs):
```

Comment:
We need to manually specify the `configs` now since we can no longer access them from the datasets. Note that this has nothing to do with the (soon to be) removed `DatasetConfig` class, but is just a list of option dictionaries, e.g. `[dict(split="train"), dict(split="test")]`.

Reply:
Do you think there's an opportunity for simplification if we relied on classes for the mocks and expose a `CONFIG` attribute, much like the `ADDITIONAL_CONFIGS` attribute of the non-prototype datasets?

Reply:
You mean like have a `class MNISTDatasetMock(DatasetMock)` that overrides a `mock_data_fn` method?

Reply:
yes, something like that. But this is the kind of optimization that we can do later